### PR TITLE
Rename statsPeriod to timeframe where appropriate (API-2880)

### LINKED ISSUE #36375: Inconsistent `statsPeriod` paramaeter
In several locations we use a parameter called `statsPeriod` in the API. Its original use was defined to adjust the period of time that statistics were calculated for the given results (specifically it would adjust the small bar graph used in the UI). Several years ago however it was changed in one endpoint to have a complete different meaning:

https://github.com/getsentry/sentry/pull/11464

This creates a few problems:

1) we have a shared API parameter with explicitly different behavior
2) the new meaning of the API parameter does not reflect the name of the parameter well (its actually a results limiter, not a 'stats period')
3) many of these endpoints are undocumented, so differing behavior becomes even worse

To correct this, we could take the following approach:

1) Deprecate all `statsPeriod` parameters - undocument them entirely
2) Add new parameters that are appropriate for different behaviors. One for the historical `statsPeriod` (or leave its singular? existing use which is correct), and one for "limit results to this period of time".
3) Alias `statsPeriod` in all endpoints to whatever this new parameter is. so it can still function and we can migrate UI code and documentation

### PR DESCRIPTION
**Pull request by rjo100**
Rename `statsPeriod` to `timeframe` for all API endpoints where appropriate. API calls with the old parameter will continue to function as normal. Also updated tests as necessary.

Fixes https://github.com/getsentry/sentry/issues/36375

### CODE REVIEW

File: src/sentry/api/endpoints/organization_group_index.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def get(self, request: Request, organization) -> Response:
         """
         stats_period = request.GET.get("groupStatsPeriod")
         try:
-            start, end = get_date_range_from_params(request.GET)
+            start, end = get_date_range_from_stats_period(request.GET)
```
Comment (rjo100) [inline src/sentry/api/endpoints/organization_group_index.py:223]: keep old logic and don't honor `timeframe` parameter

File: src/sentry/api/endpoints/organization_group_index_stats.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def get(self, request: Request, organization) -> Response:
 
         stats_period = request.GET.get("groupStatsPeriod")
         try:
-            start, end = get_date_range_from_params(request.GET)
+            start, end = get_date_range_from_stats_period(request.GET)
```
Comment (rjo100) [inline src/sentry/api/endpoints/organization_group_index_stats.py:58]: keep old logic and don't honor `timeframe` parameter

File: src/sentry/api/utils.py [hunk 3]
```diff
@@ -39,6 +39,48 @@ def default_start_end_dates(now=None, default_stats_period=MAX_STATS_PERIOD):
 
 
 def get_date_range_from_params(params, optional=False, default_stats_period=MAX_STATS_PERIOD):
```
Comment (rjo100) [inline src/sentry/api/utils.py:41]: use a wrapper function for the old logic which aliases the new parameter to the old one

File: tests/sentry/api/test_utils.py [hunk 4]
```diff
@@ -9,24 +9,33 @@
...
         assert end - datetime.timedelta(days=91) == start
 
+        start, end = get_date_range_from_params({"statsPeriod": "14h"})
+        assert end - datetime.timedelta(hours=14) == start
+
+        start, end = get_date_range_from_params({"statsPeriod": "14d"})
+        assert end - datetime.timedelta(days=14) == start
+
+        start, end = get_date_range_from_params({"statsPeriod": "60m"})
+        assert end - datetime.timedelta(minutes=60) == start
```
Comment (rjo100) [inline tests/sentry/api/test_utils.py:35]: regression testing to make sure we still honor the old parameter

File: tests/sentry/api/test_utils.py [hunk 5]
```diff
@@ -51,6 +60,11 @@ def test_no_params(self):
 
     @freeze_time("2018-12-11 03:21:34")
     def test_relative_date_range(self):
+        start, end = get_date_range_from_params({"timeframeStart": "14d", "timeframeEnd": "7d"})
```
Comment (rjo100) [inline tests/sentry/api/test_utils.py:63]: regression testing to make sure we still honor the old parameter

**Review by NisanthanNanthakumar — APPROVED** [review]

File: src/sentry/api/utils.py [hunk 6]
```diff
@@ -39,6 +39,48 @@ def default_start_end_dates(now=None, default_stats_period=MAX_STATS_PERIOD):
...
+    """
+    A wrapper function for `get_date_range_from_stats_period` that allows us
+    to alias `statsPeriod` to ensure backward compatibility.
```
[the remark this answers is not in the document]
Comment (rjo100) [reply src/sentry/api/utils.py:44]: we are explicitly not deprecating and simply undocumenting `statsPeriod`

**Pull request merged**